Subject: [PATCH] Create thumbnail if not exist.

Symptom: AP get null thumbnail from document provider.
Root Cause: Physical file has been deleted.

diff --git a/src/com/android/providers/media/MediaDocumentsProvider.java b/src/com/android/providers/media/MediaDocumentsProvider.java
index 7c3d773..84f6ce2 100644
--- a/src/com/android/providers/media/MediaDocumentsProvider.java
+++ b/src/com/android/providers/media/MediaDocumentsProvider.java
@@ -812,7 +812,13 @@ private AssetFileDescriptor openOrCreateImageThumbnailCleared(
             long id, CancellationSignal signal) throws FileNotFoundException {
         final ContentResolver resolver = getContext().getContentResolver();
 
-        ParcelFileDescriptor pfd = openImageThumbnailCleared(id, signal);
+        ParcelFileDescriptor pfd = null;
+        try {
+            pfd = openImageThumbnailCleared(id, signal);
+        } catch (FileNotFoundException e) {
+
+        }
+
         if (pfd == null) {
             // No thumbnail yet, so generate. This is messy, since we drop the
             // Bitmap on the floor, but its the least-complicated way.
@@ -900,7 +906,13 @@ private AssetFileDescriptor openOrCreateVideoThumbnailCleared(
             long id, CancellationSignal signal) throws FileNotFoundException {
         final ContentResolver resolver = getContext().getContentResolver();
 
-        AssetFileDescriptor afd = openVideoThumbnailCleared(id, signal);
+        AssetFileDescriptor afd = null;
+        try {
+            afd = openVideoThumbnailCleared(id, signal);
+        } catch (FileNotFoundException e) {
+
+        }
+
         if (afd == null) {
             // No thumbnail yet, so generate. This is messy, since we drop the
             // Bitmap on the floor, but its the least-complicated way.
